scripts/needs_testrun.py

```diff
@@ -240,10 +244,21 @@ def needs_testrun(suite: str, pr_number: int) -> bool:
     return bool(matches)
 
 
+def _get_pr_number():
+    number = os.environ.get("CIRCLE_PR_NUMBER")
+    if not number:
+        pr_url = os.environ.get("CIRCLE_PULL_REQUEST", "")
```

Comment:
ah interesting! so there are cases where the PR number is not set?

Reply:
yeah, it's only set on forked PRs https://support.circleci.com/hc/en-us/articles/360047521451-Why-is-CIRCLE-PR-NUMBER-empty-

Reply:
ah good catch!